filebeat/config/config.go

```diff
@@ -82,7 +83,15 @@ func mergeConfigFiles(configFiles []string, config *Config) error {
 			return fmt.Errorf("Failed to read %s: %s", file, err)
 		}
 
-		config.Prospectors = append(config.Prospectors, tmpConfig.Filebeat.Prospectors...)
+		if len(tmpConfig.Filebeat.Prospectors) > 0 {
+			cfgwarn.Deprecate("7.0.0", "prospectors are deprecated, Use `inputs` instead.")
+			if len(tmpConfig.Filebeat.Inputs) > 0 {
+				return fmt.Errorf("prospectors and inputs used in the configuration file, define only inputs not both")
+			}
+			config.Inputs = append(config.Inputs, tmpConfig.Filebeat.Prospectors...)
```

Comment:
Nit but code wise I would prefer if you would do here what you did above which is copying `tmpConfig.Filebeat.Prospectors` to `tmpConfig.Filebeat.Inputs` and then not do an else. Like this when the code is removed later, nothing changes in the "correct" code.

Reply:
I am not sure, my argument was to prevent people defining both **prospector** and inputs, but I understand your point I will just remove that error and just copy it. Once we hit 7.0 we can "die hard"
![landscape-1478879153-die-hard](https://user-images.githubusercontent.com/640/34994547-e636ce20-faa1-11e7-88de-faf07dc525d0.jpg)

Reply:
I have the same logic there https://github.com/elastic/beats/pull/6078/files/c2b149f7d67b082b20f8201db719ba1c01755683#diff-7a00b9fb331934c8b3bb7cf5eac57576R37, So I will remove both.

Reply:
nevermind, my previous comments, I understand your proposal.